packages/meditrak-server/src/routes/exportSurveyResponses/exportSurveyResponses.js

```diff
@@ -143,45 +149,54 @@ export async function exportSurveyResponses(req, res) {
-            survey_id: currentSurvey.id,
-            entity_id: entity.id,
-          };
-          if (startDate && endDate) {
-            surveyResponseFindConditions.submission_time = {
-              comparisonType: 'whereBetween',
-              args: [
-                [
-                  new Date(moment(startDate).subtract(1, 'day')),
-                  new Date(moment(endDate).add(1, 'day')),
-                ],
+        const surveyResponseFindConditions = {
+          survey_id: currentSurvey.id,
+        };
+        if (startDate && endDate) {
+          surveyResponseFindConditions.submission_time = {
+            comparisonType: 'whereBetween',
+            args: [
+              [
+                new Date(moment(startDate).subtract(1, 'day')),
+                new Date(moment(endDate).add(1, 'day')),
               ],
-            };
-          } else if (startDate) {
-            surveyResponseFindConditions.submission_time = {
-              comparator: '>=',
-              comparisonValue: new Date(startDate),
-            };
-          } else if (endDate) {
-            surveyResponseFindConditions.submission_time = {
-              comparator: '<=',
-              comparisonValue: new Date(endDate),
-            };
-          }
-          const surveyResponses = await models.surveyResponse.find(
-            surveyResponseFindConditions,
-            sortAndLimitSurveyResponses,
-          );
+            ],
+          };
+        } else if (startDate) {
+          surveyResponseFindConditions.submission_time = {
+            comparator: '>=',
+            comparisonValue: new Date(startDate),
+          };
+        } else if (endDate) {
+          surveyResponseFindConditions.submission_time = {
+            comparator: '<=',
+            comparisonValue: new Date(endDate),
+          };
+        }
+        const entityIdsGroup = entities.map(entity => entity.id);
+        surveyResponseFindConditions.entity_id = entityIdsGroup;
 
-          for (
-            let surveyResponseIndex = 0;
-            surveyResponseIndex < surveyResponses.length;
-            surveyResponseIndex++
-          ) {
-            await processSurveyResponse(surveyResponses[surveyResponseIndex], entity);
-          }
+        const surveyResponses = await models.surveyResponse.find(
+          surveyResponseFindConditions,
+          sortAndLimitSurveyResponses,
+        );
+        const nestedEntitiesById = keyBy(entities, 'id');
+        // Fetch all answers of all survey responses for further use in 'processSurveyResponse()'.
+        const surveyResponseIds = surveyResponses.map(response => response.id);
+        const answers = await findAnswersInSurveyResponse(
+          models,
+          surveyResponseIds,
+          {},
+          { columns: [{ [`${TYPES.SURVEY_RESPONSE}.id`]: 'survey_response.id' }], sort: [] },
+        );
+        const nestedAnswers = groupBy(answers, 'survey_response.id');
```

Comment:
`nested` is usually used when something is an array of arrays (or object of objects etc)... Here, we can make the grouping context a bit more obvious:

```suggestion
        const answersByResponseId = groupBy(answers, 'survey_response.id');
```

Reply:
Thanks Kostas, `answersByResponseId` looks good 👍